src/core/qgsrendercontext.h

```diff
@@ -152,6 +156,17 @@ class CORE_EXPORT QgsRenderContext
     /** Sets pointer to original (unsegmentized) geometry*/
     void setGeometry( const QgsAbstractGeometryV2* geometry ) { mGeometry = geometry; }
 
+    /** Set a filter feature provider used to filter the features
+     * @param ffp the filter feature provider
+     * @note not available in Python bindings
+     */
+    void setFeatureFilterProvider( const QgsFeatureFilterProvider* ffp ) { mFeatureFilterProvider = ffp; }
```

Comment:
For this I do this:
https://github.com/sbrunner/QGIS/commit/5ac6dea03c518c29b27fe125f1ed4e126e6f3424
Is it what you expected?